src/components/ListItem.jsx

```diff
@@ -23,6 +41,14 @@ export function ListItem({
 				defaultChecked={isDefaultChecked}
 			/>
 			<label htmlFor={itemId}>{name}</label>
+			<button type="button" onClick={confirmDelete}>
+				Delete
+			</button>
+			<dialog ref={dialogRef}>
+				<p>Are you sure you want to remove "{name}" from your list?</p>
+				<button onClick={handleYesClick}>Yes</button>
+				<button onClick={handleNoClick}>No</button>
+			</dialog>
```

Comment:
🔍 Excellent job implementing this Dialog element! We have an opportunity to clean it up, as well. Right now, we render a dialog for each individual ListItem. Technically, this is fine. It's unlikely to cause any problems (that I'm aware of), but it also doesn't need to be this way. We could, in theory, have exactly one dialog element, and reuse it for every list item.

Reply:
My only guess is that it needs to be its own component?